apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -346,18 +331,40 @@ export class SendMessage {
       return false;
     }
 
-    const { preference } = await this.getSubscriberTemplatePreferenceUsecase.execute(
-      GetSubscriberTemplatePreferenceCommand.create({
-        organizationId: job._organizationId,
-        subscriberId: subscriber.subscriberId,
-        environmentId: job._environmentId,
-        template,
-        subscriber,
-        tenant: job.tenant,
-      })
-    );
+    let subscriberPreference: { enabled: boolean; channels: IPreferenceChannels };
+    if (command.statelessPreferences) {
+      /*
+       * Stateless Workflow executions do not have their definitions stored in the database.
+       * Their preferences are available in the command instead.
+       *
+       * TODO: Refactor the send-message flow to better handle stateless workflows
+       */
+      const workflowPreference = GetPreferences.mapWorkflowChannelPreferencesToChannelPreferences(
+        command.statelessPreferences
+      );
+      subscriberPreference = {
+        enabled: true,
+        channels: workflowPreference,
+      };
+    } else {
+      if (!template) {
+        throw new PlatformException(`Notification template ${job._templateId} is not found`);
+      }
```

Comment:
What do you think about adding this check above so we will have more flattened code?

```suggestion
      if (!template && !command.statelessPreferences) {
        throw new PlatformException(`Notification template ${job._templateId} is not found`);
      }
```

Reply:
I prefer the `if/else` approach because it creates a clear delineation between both stateless/stateful workflows. The suggestion above leaks "stateless" mode into the stateful mode with the additional check. Summarising, I think the code is more clear as it stands currently.